Core.ApplicationServices/Users/Write/UserWriteService.cs

```diff
@@ -148,33 +153,56 @@ private Result<User, OperationError> UpdateEmail(User user, string email)
         {
-            var organizationResult = _organizationService.GetOrganization(organizationUuid);
-            if (organizationResult.Failed)
+            var oldRoles = user.GetRolesInOrganization(organization.Uuid).ToList();
+            var removeResult = RemoveRoles(user, organization, oldRoles);
+            if (removeResult.HasValue)
             {
-                return new OperationError($"Organization with uuid {organizationUuid} was not found", OperationFailure.NotFound);
+                return removeResult.Value;
             }
-            var organization = organizationResult.Value;
-            var defaultUnit = _organizationService.GetDefaultUnit(organization, user);
-            var rightsFromOtherOrganizations = user.OrganizationRights.Where(right => right.Organization.Uuid != organizationUuid);
-            var newRights = roles.Select(role => RoleToRight(role, user, organization, defaultUnit));
-            var updatedRights = rightsFromOtherOrganizations.Concat(newRights).ToList();
-            user.OrganizationRights = updatedRights;
+            var newRightsResult = AssignRoles(user, organization, roles);
+            if (newRightsResult.Failed)
+            {
+                return newRightsResult.Error;
+            }
+            var rightsFromOtherOrganizations = user.OrganizationRights.Where(right => right.Organization.Uuid != organization.Uuid).ToList();
+            rightsFromOtherOrganizations.AddRange(newRightsResult.Value);
+            user.OrganizationRights = rightsFromOtherOrganizations;
             return user;
         }
 
-        private static OrganizationRight RoleToRight(OrganizationRole role, User user, Organization organization, OrganizationUnit defaultUnit = null)
+        private Result<IEnumerable<OrganizationRight>, OperationError> AssignRoles(User user, Organization organization,
+            IEnumerable<OrganizationRole> roles)
+        {
+            var newRights = new List<OrganizationRight>();
+            foreach (var role in roles)
+            {
+                var result = _organizationRightsService.AssignRole(organization.Id, user.Id, role);
+                if (result.Failed)
+                {
+                    return new OperationError(result.Error);
+                }
+
+                newRights.Add(result.Value);
+            }
+
+            return newRights;
+        }
+
+        private Maybe<OperationError> RemoveRoles(User user, Organization organization,
+            IEnumerable<OrganizationRole> roles)
         {
-            return new OrganizationRight
+            foreach (var role in roles)
             {
-                UserId = user.Id,
-                Role = role,
-                OrganizationId = organization.Id,
-                Organization = organization,
-                DefaultOrgUnitId = defaultUnit?.Id,
-            };
+                var result = _organizationRightsService.RemoveRole(organization.Id, user.Id, role);
```

Comment:
why not use the existing removeUnitRightsByIds method
![image](https://github.com/user-attachments/assets/1c81fbf8-f467-444e-b6e7-29bf4621816c)

Reply:
I used the other methods because they operate on a user, while the the other ones operate on units